TELCODOCS-2045 - Telco core RDS 4.18 docs (#88622)
openshift/openshift-docs: modules/telco-core-cluster-common-use-model-engineering-considerations.adoc

// Module included in the following assemblies:
//
// * scalability_and_performance/telco_core_ref_design_specs/telco-core-rds.adoc

[id="telco-core-cluster-common-use-model-engineering-considerations_{context}"]
= Telco core cluster common use model engineering considerations

. Cluster workloads are detailed in "Application workloads".
. Worker nodes should run on either of the following CPUs:
.. Intel 3rd Generation Xeon (IceLake) CPUs or better when supported by {product-title}, or CPUs with the silicon security bug (Spectre and similar) mitigations turned off.
Skylake and older CPUs can experience 40% transaction performance drops when Spectre and similar mitigations are enabled.
.. AMD EPYC Zen 4 CPUs (Genoa, Bergamo, or newer) or better when supported by {product-title}.
+
[NOTE]
====
Power consumption evaluations for these CPUs are ongoing.

> You should evaluate CPU features such as per-pod power management to determine any potential impact on your requirements.

Author: Not an error: https://github.com/redhat-documentation/vale-at-red-hat/pull/917

====
.. IRQ balancing is enabled on worker nodes.
The `PerformanceProfile` CR sets `globallyDisableIrqLoadBalancing` to false.
Guaranteed QoS pods are annotated to ensure isolation as described in "CPU partitioning and performance tuning".

. All cluster nodes should have the following features:
.. Hyper-Threading enabled
.. x86_64 CPU architecture
.. Stock (non-realtime) kernel enabled
.. Nodes not configured for workload partitioning

. The balance between power management and maximum performance varies between `MachineConfigPool` groups in the cluster.
The following configurations should be consistent for all nodes in a `MachineConfigPool` group.

. Cluster scaling: see "Scalability" for more information.
.. Clusters should be able to scale to at least 120 nodes.

. CPU partitioning is configured using a `PerformanceProfile` CR and is applied to nodes on a per `MachineConfigPool` basis.
See "CPU partitioning and performance tuning" for additional considerations.
. CPU requirements for the OpenShift platform depend on the configured feature set and application workload characteristics.

> For a cluster configured according to the reference configuration, running a simulated workload of 3k pods as created by kube-burner node-density test the following CPU requirements are validated:

Author: https://github.com/redhat-documentation/vale-at-red-hat/pull/915

// Module included in the following assemblies:
//
// * scalability_and_performance/telco_core_ref_design_specs/telco-core-rds.adoc

:_mod-docs-content-type: REFERENCE
[id="telco-core-cluster-common-use-model-engineering-considerations_{context}"]
= Telco core cluster common use model engineering considerations

> . Cluster workloads are detailed in "Application workloads".

Reviewer: Should this be an unordered list, as they are not steps, from what I can tell.

. Worker nodes should run on either of the following CPUs:
.. Intel 3rd Generation Xeon (IceLake) CPUs or better when supported by {product-title}, or CPUs with the silicon security bug (Spectre and similar) mitigations turned off.
Skylake and older CPUs can experience 40% transaction performance drops when Spectre and similar mitigations are enabled.
.. AMD EPYC Zen 4 CPUs (Genoa, Bergamo, or newer) or better when supported by {product-title}.
+
[NOTE]
====
Power consumption evaluations for these CPUs are ongoing.
You should evaluate CPU features such as per-pod power management to determine any potential impact on your requirements.
====
.. IRQ balancing is enabled on worker nodes.
The `PerformanceProfile` CR sets `globallyDisableIrqLoadBalancing` to false.
Guaranteed QoS pods are annotated to ensure isolation as described in "CPU partitioning and performance tuning".

. All cluster nodes should have the following features:
.. Hyper-Threading enabled
.. x86_64 CPU architecture
.. Stock (non-realtime) kernel enabled

> .. Nodes not configured for workload partitioning

Reviewer: My ignorance: Is there a positive way to word the last bullet to be consistent?

```suggestion
.. Nodes configured for <blah> partitioning
```

Author: No, it's a critical feature that must be disabled. But I take your point and will rework for parallelism

* The balance between power management and maximum performance varies between machine config pools in the cluster.
The following configurations should be consistent for all nodes in a machine config pools group.

> * Cluster scaling. See "Scalability" for more information.

Reviewer 2: Based on the above sentence that says "The following", is this "cluster scaling" line supposed to be a second level bullet?

> . The balance between power management and maximum performance varies between `MachineConfigPool` groups in the cluster.
> The following configurations should be consistent for all nodes in a `MachineConfigPool` group.

Reviewer: I don't think the MCO team (or our docs) uses _MachineConfigPool groups_. 

```suggestion
. The balance between power management and maximum performance varies between machine config pools in the cluster.
The following configurations should be consistent for all nodes in a machine config pools group.
```

> . Cluster scaling: see "Scalability" for more information.
> .. Clusters should be able to scale to at least 120 nodes.

Reviewer: Not thrilled with the single sub bullet.

```suggestion
. Clusters should be able to scale to at least 120 nodes. See "Scalability" for more information.
```

> . Cluster scaling: see "Scalability" for more information.
> .. Clusters should be able to scale to at least 120 nodes.

Reviewer: ```suggestion
. Cluster scaling. See "Scalability" for more information.
.. Clusters should be able to scale to at least 120 nodes.
```

> . CPU partitioning is configured using a `PerformanceProfile` CR and is applied to nodes on a per `MachineConfigPool` basis.

Reviewer: Need a noun after the object reference or get rid of the mark-up.:

```suggestion
. CPU partitioning is configured using a `PerformanceProfile` CR and is applied to nodes on a per machine config pool basis.
```

* CPU requirements for the OpenShift platform depend on the configured feature set and application workload characteristics.

> For a cluster configured according to the reference configuration, running a simulated workload of 3k pods as created by the kube-burner node-density test the following CPU requirements are validated:

Reviewer 2: Should 3k be spelled out as 3000?

For a cluster configured according to the reference configuration, running a simulated workload of 3k pods as created by kube-burner node-density test the following CPU requirements are validated:

> .. The minimum number of reserved CPUs for supervisor and worker nodes is 2 CPUs (4 hyper-threads) per NUMA node.

Reviewer: Is _supervisor node_ a telco term?

Author: Yes, they orchestrate where Telco CNF workloads go in the telco network. Can be control plane or not.

> Nodes with large numbers of pods or other resources may require additional reserved CPUs.

Reviewer: ```suggestion
Nodes with large numbers of pods or other resources might require additional reserved CPUs.
```

// Module included in the following assemblies:
//
// * scalability_and_performance/telco_core_ref_design_specs/telco-core-rds.adoc

:_mod-docs-content-type: REFERENCE
[id="telco-core-cluster-common-use-model-engineering-considerations_{context}"]
= Telco core cluster common use model engineering considerations

* Cluster workloads are detailed in "Application workloads".
* Worker nodes should run on either of the following CPUs:
** Intel 3rd Generation Xeon (IceLake) CPUs or better when supported by {product-title}, or CPUs with the silicon security bug (Spectre and similar) mitigations turned off.
Skylake and older CPUs can experience 40% transaction performance drops when Spectre and similar mitigations are enabled.
** AMD EPYC Zen 4 CPUs (Genoa, Bergamo, or newer) or better when supported by {product-title}.
+
[NOTE]
====

> Power consumption evaluations for these CPUs are ongoing.

Reviewer 3: We just discovered an issue, this needs an update. Per pod power management is currently not working for AMD.

Author: Added a note

Reviewer 3: Read the whole note - it does not make sense now. Keep just the last sentence.
